### ISSUE TITLE
Relax HTTPS requirement for Opencast and Meili URLs

### ISSUE THREAD

**Issue by LukasKalbertodt** [elan-ev/tobira#400]
Our configuration is fairly strict about the use of HTTP/HTTPS in the URLs used to communicate with Opencast or Meili. If the IP address does not resolve to an loopback address, we forbid HTTP. But this might be a bit too strict. With docker containers, for example, there could be non-loopback addresses that are still local, i.e. traffic never leaves the machine. Also, communicating between VMs in a data center might be safe as well.

Our assumption was: you will have HTTPS for Opencast anyway, since users are supposed to use it (and there you better have HTTPS!). But, the TLS stuff could very well be done by a reverse proxy or load balancer. And of course it's understandable that Tobira↔OC communication should maybe not go through that.

At the same time, we don't want to make it too easy to accidentally specify HTTP URLs. When manually writing URLs, it happens to me somewhat regularly. So maybe actually have a flag that you can explicitly specify?

**Comment by lkiesow** [comment]
Are we talking about admin settings? If so, I think it's the administrator's responsibility and I would let them specify whatever they want.

**Comment by LukasKalbertodt** [comment]
Yes, the settings in `config.toml`. I get what you're saying, but I'm generally against the stance "$human should know what they are doing, so don't bother having any checks or warnings". Humans making mistakes is a given. Of course, there is a balance to strike here, as usual. 

Anyway, we will certainly relax it, no doubts about that. The current check is likely overly cautious :P

**Comment by lkiesow** [comment]
Browsers will already warn you if you set a non-secure URL or things might now even work. But as you pointed out, there is still a valid use-case for HTTP. More than that, we are not talking about average users in this case, but we expect trained personal to handle this. That's why I'm hesitant about enforcing this, when there is not a big benefit due to what I think is a low likelihood of operators accidentally setting insecure locations and no-one noticing.
